src/hooks/useSafes.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react'
+import type { ChainInfo, AddressEx } from '@safe-global/safe-gateway-typescript-sdk'
+import { getBalances } from '@safe-global/safe-gateway-typescript-sdk'
+
+import useChainId from '@/hooks/useChainId'
+import useChains from '@/hooks/useChains'
+import { useAppSelector } from '@/store'
+import { selectAllAddedSafes } from '@/store/addedSafesSlice'
+import useAsync from './useAsync'
+
+type SafeListItemDetails = {
+  chain: ChainInfo
+  safeAddress: string
+  fiatBalance?: string
+  threshold: number
+  owners: AddressEx[]
+}
+
+const sortChainsByCurrentChain = (chains: ChainInfo[], currentChainId: string): ChainInfo[] => {
+  const currentChain = chains.find(({ chainId }) => chainId === currentChainId)
+  const otherChains = chains.filter(({ chainId }) => chainId !== currentChainId)
+  return currentChain ? [currentChain, ...otherChains] : chains
+}
+
+export const useWatchedSafes = (): [SafeListItemDetails[], Error | undefined, boolean] => {
+  const currentChainId = useChainId()
+  const { configs } = useChains()
+  const watchedSafes = useAppSelector(selectAllAddedSafes)
+  const chains = useMemo(() => sortChainsByCurrentChain(configs, currentChainId), [configs, currentChainId])
+
+  const [allWatchedSafesWithBalances, error, loading] = useAsync<SafeListItemDetails[]>(
+    () => {
+      let watchedSafesOnAllChains: SafeListItemDetails[] = []
+      for (const chain of chains) {
+        const watchedSafesOnChain = watchedSafes[chain.chainId] ?? {}
+        const watchedSafesAdressesOnChain = Object.keys(watchedSafesOnChain)
+        const watchedSafesWithChain = watchedSafesAdressesOnChain.map((safeAddress) => {
+          const { threshold, owners } = watchedSafesOnChain[safeAddress]
+          return { safeAddress, chain, threshold, owners }
+        })
+        watchedSafesOnAllChains = [...watchedSafesOnAllChains, ...watchedSafesWithChain]
+      }
+      const promises = watchedSafesOnAllChains.map(async ({ safeAddress, chain, threshold, owners }) => {
+        const fiatBalance = await getBalances(chain.chainId, safeAddress, 'USD').then((result) => result.fiatTotal)
```

Comment:
USD shouldn't be hardcoded. There's a redux selector for the user-preferred currency.

Reply:
Overall, I wouldn't fetch balances in this hook. They essentially make the entire list wait. I would instead fetch the balances separately and then pass them separately to the corresponding components.

Reply:
yeah that makes sense. I moved the balance request to SafeListItem